src/overlays/actors/ovl_En_Mm2/z_en_mm2.c

```diff
@@ -43,33 +48,292 @@ static ColliderCylinderInit D_80AAFAC0 = {
     },
     { 18, 63, 0, { 0, 0, 0 } },
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Mm2/func_80AAEE50.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Mm2/func_80AAEF70.s")
+typedef struct {
+    /* 0x00 */ AnimationHeader* animation;
+    /* 0x04 */ f32 playSpeed;
+    /* 0x08 */ u8 mode;
+    /* 0x0C */ f32 morphFrames;
+} unknownStruct; // size = 0x10
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Mm2/EnMm2_Init.s")
+static unknownStruct sAnimations[] = {
+    0x06000718, 1.0f, 0, -7.0f, 0x06006940, -1.0f, 2, -7.0f,  0x06006C50, 1.0f, 0, -7.0f,  0x06006940, 1.0f, 2, -7.0f,
+    0x06000468, 1.0f, 0, -7.0f, 0x060073A0, 1.0f,  0, -12.0f, 0x06008060, 1.0f, 0, -12.0f,
+};
+
+static InitChainEntry sInitChain[] = {
+    ICHAIN_F32(uncullZoneForward, 4000, ICHAIN_STOP),
+};
+
+void EnMm2_ChangeAnimation(EnMm2* this, s32 animationIndex, s32* previousAnimation) {
+    f32 phi_f0;
+
+    if ((*previousAnimation < 0) || (animationIndex == *previousAnimation)) {
+        phi_f0 = 0.0f;
+    } else {
+        phi_f0 = sAnimations[animationIndex].morphFrames;
+    }
+
+    if (sAnimations[animationIndex].playSpeed >= 0.0f) {
+        Animation_Change(&this->skelAnime, sAnimations[animationIndex].animation, sAnimations[animationIndex].playSpeed,
+                         0.0f, (f32)Animation_GetLastFrame(sAnimations[animationIndex].animation),
+                         sAnimations[animationIndex].mode, phi_f0);
+    } else {
+        Animation_Change(&this->skelAnime, sAnimations[animationIndex].animation, sAnimations[animationIndex].playSpeed,
+                         (f32)Animation_GetLastFrame(sAnimations[animationIndex].animation), 0.0f,
+                         sAnimations[animationIndex].mode, phi_f0);
+    }
+    *previousAnimation = animationIndex;
+}
+
+void func_80AAEF70(EnMm2* this, GlobalContext* globalCtx) {
+    if ((gSaveContext.eventChkInf[9] & 0xF) != 0xF) {
+        this->actor.textId = 0x6086;
+        return;
+    }
```

Comment:
at first glance it seems like this function can be improved with else ifs and elses. might be overlooking something tho